src/Features/ERDDAP/Platform/Observations/CurrentConditions/data_card.tsx

```diff
@@ -1,95 +1,153 @@
+/**
+ * Generalized single time series data card
+ */
 import React from "react"
 import { Link } from "react-router-dom"
 import { Card, CardBody, CardHeader, Col } from "reactstrap"
 
 import { SmallTimeSeriesChart } from "components/Charts"
-import { paths } from "Shared/constants"
-
 import { naturalBounds } from "Shared/dataTypes"
 import { round } from "Shared/math"
+import { DataTimeSeries, ReadingTimeSeries } from "Shared/timeSeries"
 import { convertUnit } from "Shared/unitConversion"
-import { urlPartReplacer } from "Shared/urlParams"
+import { useUnitSystem } from "Features/Units"
 import { UnitSystem } from "Features/Units/types"
 import { converter } from "Features/Units/Converter"
 
-import { PlatformFeatureWithDatasets, PlatformDataset } from "../../../types"
+import { useDataset } from "../../../hooks"
+import { PlatformFeature, PlatformTimeSeries } from "../../../types"
+
+import { cardProps, cardUrl } from "./common_card"
+import { NoDataCard } from "./no_data_card"
+import { LoadingDataCard } from "./loading_card"
+import { ErrorDataCard } from "./error_card"
 
 interface DataCardProps {
-  platform: PlatformFeatureWithDatasets
+  /** Data Type standard names to display */
   data_types: string[]
-  unit_system: UnitSystem
+  /** Platform */
+  platform: PlatformFeature
 }
 
-const cardProps = {
-  md: "4",
-  sm: "6",
-  style: {
-    paddingTop: "1rem",
-  },
-}
+/**
+ * Filters data types for a platform and fetches appropriate one
+ * to display based on preferential order and what is up to date
+ * @param data_types Data Type standard names to display
+ * @param platform Platform to display
+ */
+export const DataCard: React.SFC<DataCardProps> = ({ data_types, platform }) => {
+  const aDayAgo = new Date()
+  aDayAgo.setDate(aDayAgo.getDate() - 1)
 
-export const DataCard: React.SFC<DataCardProps> = ({ platform, data_types, unit_system }) => {
-  let filtered_datasets: PlatformDataset[] = []
+  let filteredTimeSeries: PlatformTimeSeries[] = []
 
-  data_types.forEach((data_type) => {
+  data_types.forEach((dataType) => {
```

Comment:
Codacy found an issue: [Identifier 'data_types' is not in camel case.](https://app.codacy.com/manual/gmri/Neracoos-1-Buoy-App/pullRequest?prid=6166252)